tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -4364,7 +4364,7 @@ bool MIOpenSupport::DoPoolBackward(
   DeviceMemory<uint8> workspace;
   size_t workspace_size_in_bytes = 0;
   auto status = wrap::miopenPoolingGetWorkSpaceSizeV2(
-      pooling_desc.handle(), dest_desc.handle(), &workspace_size_in_bytes);
+      pooling_desc.handle(), src_desc.handle(), &workspace_size_in_bytes);
```

Comment:
If I understand it right, it should use the original `yDesc`, which means the original `dest_desc` is correct. Documentation [here](https://rocmsoftwareplatform.github.io/MIOpen/doc/html/pooling.html#miopenpoolinggetworkspacesizev2).

Reply:
Okay, I see the problem now.
https://rocmsoftwareplatform.github.io/MIOpen/doc/html/pooling.html#miopenpoolingforward

do_backward: Boolean to toggle save data in workspace for backwards pass (input) 

`miopenPoolingBackward` can work with the content of `output_data` instead of `dest2`, but it also needs some private data that's computed during the `miopenPoolingForward` call and deposited into `workspace`. That's why we need that call.

Reply:
Nice, this does clarify my doubts. 

I'm seeing you implements caching for pooling, which is essentially trading memory for speed. I think that's a good shot for solving the problem. On the other hand, I never see pooling speed being an issue in model execution, so I feel unsure about this optimization. 

I'm fine with you reverting the implementation back to the point of full CI pass, and attach the above response as a comment so that someone else doesn't have to go through the rabbit hole as we do here.